src/libraries/System.Runtime.Extensions/tests/System/EnvironmentTests.cs

```diff
@@ -333,7 +332,8 @@ public void FailFast_ExceptionStackTrace_InnerException()
 
         [Fact]
         [PlatformSpecific(TestPlatforms.AnyUnix | TestPlatforms.Browser)]  // Tests OS-specific environment
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/36896", TestPlatforms.iOS | TestPlatforms.tvOS | TestPlatforms.MacCatalyst)]
+        [SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.tvOS, "Not valid on iOS or tvOS")]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/36896", TestPlatforms.MacCatalyst)]
```

Comment:
Do you expect that this test can be fixed to be valid on Catalyst? Catalyst is generally like iOS.

Reply:
Yeah Catalyst has the same behavior as iOS here. We can actually enable parts of the test since we just use a different value for `SpecialFolder.Personal` / `SpecialFolder.MyDocuments`. Will push to the PR.

Reply:
Looking more closely, it looks like `Environment.GetEnvironmentVariable("Home")` is `null` or empty on iOSSimulator and tvOSSimulator. However, on MacCatalyst, it seems to be `/Users/mdhwang` and the tests expected paths are 
`Environment.GetFolderPath(Environment.SpecialFolder.Personal)` evaluates to `/Users/mdhwang/Documents`
`Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments)` evaluates to `/Users/mdhwang/Documents`
`Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)` evaluates to `/Users/mdhwang`

@filipnavara 
> GetFolderPath_Unix_PersonalIsHomeAndUserProfile is not valid for iOS, tvOS

Reply:
Hm I tried with preview7 on iOSSimulator and `Environment.GetEnvironmentVariable("HOME")` returned a path there. did you test with `Home` or `HOME`?

Reply:
> Could you elaborate on why its not valid for iOS/tvOS?

On iOS:
- Personal/MyDocuments (same enum value) returns value of `NSSearchPathDirectory.NSDocumentDirectory`
- `UserProfile` return the `HOME` environment variable. I remember testing it but I don't remember the value anymore, my previous comment would imply that it was `null`

On tvOS:
- `NSSearchPathDirectory.NSDocumentDirectory` is read-only so Personal/MyDocuments intentionally does NOT return it. At least that was the situation on old Mono. Apparently the code was not ported for .NET 6 and it's still tracked as an issue - https://github.com/dotnet/runtime/issues/34007

Reply:
> did you test with `Home` or `HOME`?
I used `HOME`
```
[Fact]
[PlatformSpecific(TestPlatforms.AnyUnix | TestPlatforms.Browser)]  // Tests OS-specific environment
public void GetFolderPath_Unix_PersonalIsHomeAndUserProfile()
{
    Console.WriteLine("Mitchell Hwang");
    Console.WriteLine(Environment.GetEnvironmentVariable("HOME"));
    Console.WriteLine(Environment.GetFolderPath(Environment.SpecialFolder.Personal));
    Console.WriteLine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments));
    Console.WriteLine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile));
    Assert.Equal(Environment.GetEnvironmentVariable("HOME")+"/Documents", Environment.GetFolderPath(Environment.SpecialFolder.Personal));
    Assert.Equal(Environment.GetEnvironmentVariable("HOME")+"/Documents", Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments));
    Assert.Equal(Environment.GetEnvironmentVariable("HOME"), Environment.GetFolderPath(Environment.SpecialFolder.UserProfile));
}
```

The most recent run locally on preview 6 was
`Assert.Equal() Failure\nExpected: (null)\nActual:   /Users/mdhwang/Library/Developer/CoreSimulator/Devices/71D00C72-A1BD-4EF7-87EA-B22842EC1650/data/Containers/Data/Application/E3609B50-9E46-4864-A413-A1C506C5295D/Documents`